DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  ProfileViewModel.swift
+//  DigitalLibrary
+//
+//  Created by Ognyan Yonchev on 16.02.24.
+//
+
+import SwiftUI
+
+final class ProfileViewModel: ObservableObject {
+    //@ObservedObject private var appRootManager: AppRootManager
+    let authenticationProvider: AuthenticationProvidable
+    let usersProvider: UserProvidable
+
+    @Published var user: UserModel?
+    @Published var editingField: ProfileField? = nil
+
+    @Published var email: String = ""
+    @Published var firstName: String = ""
+    @Published var lastName: String = ""
+    @Published var phone: String = ""
+
+    init(//appRootManager: AppRootManager,
+         authenticationProvider: AuthenticationProvidable,
+         usersProvider: UserProvidable) {
+        //self.appRootManager = appRootManager
+        self.authenticationProvider = authenticationProvider
+        self.usersProvider = usersProvider
+    }
+
+    func fetchCurrentUser() {
+        Task {
+            do {
+                guard let fetchedUser = await usersProvider.getCurrentUser() else {
+                    // Handle the case where no user is fetched
+                    print("No user data was fetched.")
```

Comment:
Same here, we want to show alert with the error

Reply:
ok